ludwig/models/trainer.py

```diff
@@ -1355,8 +1358,9 @@ def callback(self, fn, coordinator_only=True):
 
 class RemoteTrainer(Trainer):
     def __init__(self, gpus=None, gpu_memory_limit=None, allow_parallel_threads=True, **kwargs):
-        horovod = initialize_horovod()
-        super().__init__(horovod=horovod, **kwargs)
+        # horovod = initialize_horovod()
```

Comment:
Noticed that we initialize horovod twice, once in `train_fn` and once in `RemoteTrainer`. Not sure if this is intentional. If not, I can also remove this from the legacy trainer.

Reply:
The only thing here is you will need to make sure you plumb `horovod` from the caller that creates the `RemoteTrainer` to here. I think it's probably safer to leave this as is (calling `hvd.init()` multiple times is safe).

Reply:
I'd added the plumbing in the `RemoteTrainer` initialization inside the `train_fn`, but hadn't added it yet inside `legacy_train_fn`.

I can remove this if unnecessary.

Reply:
Let's try doing the plumbing for the legacy trainer as well, just need to put it in the `executable_kwargs` here:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/backend/ray.py#L293

Reply:
Also, we should delete the commented out code.

Reply:
@tgaddair double checking this, looks like hvd is independently inside `legacy_train_fn` that the `RayTrainExecutor` executes. Given that, do we need to initialize hvd inside `RayLegacyTrainer` and then pass it to `executable_kwargs`?